src/fswAlgorithms/attGuidance/prescribedRot2DOF/_UnitTest/test_prescribedRot2DOF.py

```diff
@@ -0,0 +1,323 @@
+    PrescribedRot2DOFConfig.phiDDotMax = phiDDotMax
+    PrescribedRot2DOFConfig.r_FM_M = np.array([1.0, 0.0, 0.0])                  # [m] Position of the F frame origin relative to the M frame origin in M frame components
+    PrescribedRot2DOFConfig.rPrime_FM_M = np.array([0.0, 0.0, 0.0])             # [m/s] B frame time derivative of r_FM_M in M frame components
+    PrescribedRot2DOFConfig.rPrimePrime_FM_M = np.array([0.0, 0.0, 0.0])        # [m/s^2] B frame time derivative of rPrime_FM_M in M frame components
+    PrescribedRot2DOFConfig.omega_FM_F = np.array([0.0, 0.0, 0.0])              # [rad/s] Angular velocity of frame F relative to frame M in F frame components
+    PrescribedRot2DOFConfig.omegaPrime_FM_F = np.array([0.0, 0.0, 0.0])         # [rad/s^2] B frame time derivative of omega_FB_F in F frame components
+    PrescribedRot2DOFConfig.sigma_FM = np.array([0.0, 0.0, 0.0])                # MRP attitude of frame F relative to frame M
+
+    # Add test module to runtime call list
+    unitTestSim.AddModelToTask(unitTaskName, PrescribedRot2DOFWrap, PrescribedRot2DOFConfig)
+
+    # Create the prescribedRot2DOF input message
+    thetaDot_Ref = 0.0  # [rad/s]
+    HingedRigidBodyMessageData1 = messaging.HingedRigidBodyMsgPayload()
+    HingedRigidBodyMessageData2 = messaging.HingedRigidBodyMsgPayload()
+    HingedRigidBodyMessageData1.theta = thetaRef1a
+    HingedRigidBodyMessageData2.theta = thetaRef2a
+    HingedRigidBodyMessageData1.thetaDot = thetaDot_Ref
+    HingedRigidBodyMessageData2.thetaDot = thetaDot_Ref
+    HingedRigidBodyMessage1 = messaging.HingedRigidBodyMsg().write(HingedRigidBodyMessageData1)
+    HingedRigidBodyMessage2 = messaging.HingedRigidBodyMsg().write(HingedRigidBodyMessageData2)
+    PrescribedRot2DOFConfig.spinningBodyRef1InMsg.subscribeTo(HingedRigidBodyMessage1)
+    PrescribedRot2DOFConfig.spinningBodyRef2InMsg.subscribeTo(HingedRigidBodyMessage2)
+
+    # Set up message data recording logging on the test module output message to get access to it
+    dataLog = PrescribedRot2DOFConfig.prescribedMotionOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, dataLog)
+
+    # Set up module variable data recording
+    unitTestSim.AddVariableForLogging(PrescribedRot2DOFWrap.ModelTag + ".phi", testProcessRate, 0, 0, 'double')
+    unitTestSim.AddVariableForLogging(PrescribedRot2DOFWrap.ModelTag + ".phiAccum", testProcessRate, 0, 0, 'double')
+
+    # Initialize the simulation
+    unitTestSim.InitializeSimulation()
+
+    # Calculate the two reference PRVs for the first attitude maneuver
+    prv_F0M_a = [thetaRef1a * rotAxis1_M[0], thetaRef1a * rotAxis1_M[1], thetaRef1a * rotAxis1_M[2]]
+    prv_F1F0_a = [thetaRef2a * rotAxis2_F1[0], thetaRef2a * rotAxis2_F1[1], thetaRef2a * rotAxis2_F1[2]]
+
+    # Calculate a single reference PRV for the first attitude maneuver and the associated MRP attitude
+    if (thetaRef1a == 0 and thetaRef2a == 0):  # Prevent a (0,0,0) error using rbk.addPRV()
```

Comment:
remove brackets, they are not needed here in python

Reply:
Done.